TheThingsNetwork/lorawan-stack
# Support batch deletion of gateways

#### Summary



This is a follow up to #6455 and the final backend part of https://github.com/TheThingsIndustries/product-management/issues/6.

This is blocked on #6455 

#### Changes



- Add new RPC and message types.
- Add registry and store implementations.
- Add tests

#### Testing



1. Create some gateways
2. Login as a user with read-only rights.
3. Try to delete gateways using `ttn-lw-cli gtw batch-delete <gtw-id-1> <gtw-id-2>...`. This should error with `insufficient rights`.
4. Now login as a user with delete rights.
5. Delete gateways `ttn-lw-cli gtw batch-delete <gtw-id-1> <gtw-id-2>...`. This should succeed.

##### Regressions



This is new functionality. No regressions expected.

#### Notes for Reviewers



This targets `feature/batch-gateway-rights` and should be merged after it.

#### Checklist
